### PR TITLE
sambacc: extend rados pseudo-uris to include getting mon config keys

### PR DESCRIPTION
**Pull request by phlogistonjohn**
There are cases where certain configuration items may not be stored in the Ceph cluster's  rados pool but rather in the Ceph MON's configuration key store. This can be accessed via a mon command. Add support for new pseudo uri's formed like:
  rados:mon-config-key:foo/bar/baz

Instead of fetching an item from a rados pool, this will try to fetch a key in a manner equivalent to running the following CLI command:
  `ceph config-key get foo/bar/baz`

This idea is borrowed from the ceph iscsi support.

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
I will clear the DNM label once I have done a 2nd round of "clean" testing on a real ceph cluster.

File: sambacc/rados_opener.py [hunk 1]
```diff
@@ -171,6 +177,25 @@ def writelines(self, ls: typing.Iterable[typing.Any]) -> None:
...
+    )
+    with rados_api.Rados(conffile=rados_api.Rados.DEFAULT_CONF_FILES) as rc:
+        ret, out, err = rc.mon_command(mcmd, b"")
+        if ret == 0:
+            # We need to return a file like object. Since we are handed just
+            # bytes from this api, use BytesIO to adapt it to something valid.
+            return io.BytesIO(out)
+        # ensure ceph didn't send us a negative errno
+        ret = ret if ret > 0 else -ret
+        msg = f"failed to get mon config key: {key!r}: {err}"
```
[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply sambacc/rados_opener.py:195]: I like getting the repr as it helps reveal any "weird" characters in strings and/or reveals if an unexpected type.

File: sambacc/rados_opener.py [hunk 2]
```diff
@@ -171,6 +177,25 @@ def writelines(self, ls: typing.Iterable[typing.Any]) -> None:
...
+        }
+    )
+    with rados_api.Rados(conffile=rados_api.Rados.DEFAULT_CONF_FILES) as rc:
+        ret, out, err = rc.mon_command(mcmd, b"")
+        if ret == 0:
+            # We need to return a file like object. Since we are handed just
+            # bytes from this api, use BytesIO to adapt it to something valid.
+            return io.BytesIO(out)
+        # ensure ceph didn't send us a negative errno
+        ret = ret if ret > 0 else -ret
```
[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply sambacc/rados_opener.py:194]: Heh. I keep mistakenly thinking `abs` is in the math module. If you don't mind I don't think I'll change the PR for just this but thank you for pointing that out and reminding me `abs` doesn't require an additional import.

[the remark this answers is not in the document]
Comment (synarete) [reply sambacc/rados_opener.py:194]: Sure, just a matter of style.

**Review by synarete — APPROVED** [review]
lgtm

**Review by anoopcs9 — APPROVED** [review]

**Pull request merged**